src/Mod/TechDraw/App/DimensionFormatter.cpp

```diff
@@ -308,23 +308,60 @@ std::string DimensionFormatter::getFormattedDimensionValue(const int partial) co
+// specifiers '%w', which works as %f but without trailing zeros
+// and '%r', which round the value to the given precision.
+QString DimensionFormatter::formatValueToSpec(const double value, QString formatSpecifier) const
 {
     QString formattedValue;
-    if (formatSpecifier.contains(QRegularExpression(QStringLiteral("%.*[wW]")))) {
-        QString fs = formatSpecifier;
-        fs.replace(QRegularExpression(QStringLiteral("%(.*)w")), QStringLiteral("%\\1f"));
-        fs.replace(QRegularExpression(QStringLiteral("%(.*)W")), QStringLiteral("%\\1F"));
-        formattedValue = QString::asprintf(fs.toStdString().c_str(), value);
-        // First, try to cut trailing zeros, if AFTER decimal dot there are nonzero numbers
-        // Second, try to cut also decimal dot and zeros, if there are just zeros after it
-        formattedValue.replace(QRegularExpression(QStringLiteral("([0-9][0-9]*\\.[0-9]*[1-9])00*$")), QStringLiteral("\\1"));
-        formattedValue.replace(QRegularExpression(QStringLiteral("([0-9][0-9]*)\\.0*$")), QStringLiteral("\\1"));
-    } else {
+
+    auto format = [](QString f, double value){
+        return QString::asprintf(f.toStdString().c_str(), value);
+    };
+
+    QRegularExpression wrRegExp(QStringLiteral("%(?<dec>.*)(?<spec>[wWrR])"));
+    QRegularExpressionMatch wrMatch = wrRegExp.match(formatSpecifier);
+
+    if (! wrMatch.hasMatch()) {
         if (isNumericFormat(formatSpecifier)) {
-            formattedValue = QString::asprintf(formatSpecifier.toStdString().c_str(), value);
+            formattedValue = format(formatSpecifier, value);
         }
+        return formattedValue;
+    }
+
+    QString spec = wrMatch.captured(QStringLiteral("spec")).toLower();
+    QString dec = wrMatch.captured(QStringLiteral("dec"));
+
+    if (spec == QStringLiteral("w")) {
+        formattedValue = format(QStringLiteral("%") + dec + QStringLiteral("f"), value);
+        // First, cut trailing zeros
+        while(formattedValue.back().toLatin1() == '0')
+        {
+            formattedValue.chop(1);
+        }
+        // Second, try to cut also decimal dot
+        if(formattedValue.back().toLatin1() == '.')
+        {
+            formattedValue.chop(1);
+        }
+    }
+    else if (spec == QStringLiteral("r")) {
+        // round the value to the given precision
+        double rounder = dec.toDouble();
+        double roundValue = std::ceil(value / rounder) * rounder;
+        // format the result with the same decimal count than the rounder
+        int dotIndex = dec.indexOf(QStringLiteral("."));
+        int nDecimals = 0;
+        if (dotIndex >= 0){
+            // remove trailing zeros to avoid decimal overwriting
+            while(dec.back().toLatin1() == '0')
+            {
+                dec.chop(1);
+            }
+            dotIndex = dec.indexOf(QStringLiteral("."));
+            nDecimals = dec.size() - dotIndex - 1;
+        }
```

Comment:
```suggestion
        while(dec.endsWith("0"))  {
            dec.chop(1);
        }
        int dotIndex = dec.indexOf(QStringLiteral("."));
        int nDecimals = dec.size() - dotIndex - 1;
```

Reply:
This doesn't work if there is no dot i.e. `dotIndex = -1` but I'll keep the concept.

Reply:
so `dotIndex` computation is needed above `dec.contains(".")`. In order to avoid more string computation I keep my condition `dotIndex >= 0`

Reply:
It's not untrue. But readability is probably more important than 0,00001ms 🤏 performance gain. I mean, we could get higher performance using const char* everywhere.

In any case, it's fine with me as it is.